src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/diff/AlertsUpdatedEventService.kt

```diff
@@ -0,0 +1,24 @@
+package uk.gov.justice.digital.hmpps.prisonersearch.services.diff
+
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.prisonersearch.model.Prisoner
+import uk.gov.justice.digital.hmpps.prisonersearch.services.HmppsDomainEventEmitter
+
+@Service
+class AlertsUpdatedEventService(
+  private val domainEventEmitter: HmppsDomainEventEmitter,
+) {
+  fun generateAnyEvents(
+    previousPrisonerSnapshot: Prisoner?,
+    prisoner: Prisoner
+  ) {
+    val previousAlertList = previousPrisonerSnapshot?.alerts?.map { it.alertCode }?.toSet() ?: emptySet()
+    val alertList = prisoner.alerts?.map { it.alertCode }?.toSet() ?: emptySet()
```

Comment:
trivial point which looks slightly odd - your list variables here are actually sets

Reply:
Yeah, I'll rename in next pr